bazel/envoy_build_system.bzl

```diff
@@ -681,7 +681,4 @@ def envoy_select_boringssl(if_fips, default = None):
 
 # Selects the part of QUICHE that does not yet work with the current CI.
 def envoy_select_quiche(xs, repository = ""):
-    return select({
-        repository + "//bazel:enable_quiche": xs,
-        "//conditions:default": [],
-    })
+    return xs
```

Comment:
Do you plan on keeping the select around for the long term, or is now a good time to clean up all call sites?

More generally, are there situations where folks will want to build Envoy without QUIC in the future? My intuition says no, because we will probably become increasingly dependent on QUICHE, e.g. if there are utilities in there that we want in Envoy core.

Reply:
> Do you plan on keeping the select around for the long term, or is now a good time to clean up all call sites?
> 
I do not plan to keep this select for long, if this PR is merged, I'll remove the select in a separate PR, that would be a very mechanical change.

> More generally, are there situations where folks will want to build Envoy without QUIC in the future? My intuition says no, because we will probably become increasingly dependent on QUICHE, e.g. if there are utilities in there that we want in Envoy core.

I'm not sure but my intuition is also no.

Reply:
OK, great, let's do this in a followup.